### PR TITLE
async user download

### PR DESCRIPTION
**Pull request by snopoke**
https://manage.dimagi.com/default.asp?250596

This is a pretty quick and dirty solution.

### CODE REVIEW

File: corehq/apps/users/urls.py [hunk 1]
```diff
@@ -118,6 +118,10 @@
     url(r'^commcare/upload/poll/(?P<download_id>[0-9a-fA-Z]{25,32})/$',
         user_upload_job_poll, name='user_upload_job_poll'),
     url(r'^commcare/download/$', download_commcare_users, name='download_commcare_users'),
+url(r'^commcare/download/status/(?P<download_id>[0-9a-fA-Z]{25,32})/$', DownloadUsersStatusView.as_view(),
```
Comment (benrudolph) [inline corehq/apps/users/urls.py]: nit: could be just the diff but the spacing looks weird

**Review by benrudolph — APPROVED** [review]
nice, looks good to unblock them. i think it won't be hard to add user's export and/or add IMEI to the commcare-user case export as well

File: corehq/apps/style/templates/style/soil_status_full.html [hunk 2]
```diff
@@ -13,7 +13,11 @@
                             $("#display_{{ download_id }}").html(data);
                         },
                         error: function(data) {
-                            $("#display_{{ download_id }}").html('<p class="alert alert-danger">{{ error_text }}</p>');
+                            if (data.responseText !== undefined && data.responseText !== ''){
+                                $("#display_{{ download_id }}").html('<p class="alert alert-danger">' + data.responseText + '</p>');
```
Comment (sravfeyn) [inline corehq/apps/style/templates/style/soil_status_full.html]: nit: could reuse this long line 

```
if (data.responseText !== undefined && data.responseText !== ''){
     var message = data.responseText;
} else {
     var message = {{ error_text }};
}
$("#display_{{ download_id }}").html('<p class="alert alert-danger">' + message + '</p>');
```

**Review by sravfeyn — APPROVED** [review]

File: corehq/apps/users/views/mobile/users.py [hunk 3]
```diff
@@ -1152,43 +1148,44 @@ def errors(self):
     return render(request, template, context)
 
 
-@require_can_edit_commcare_users
-def download_commcare_users(request, domain):
-    response = HttpResponse(content_type=Format.from_format('xlsx').mimetype)
-    response['Content-Disposition'] = 'attachment; filename="%s_users.xlsx"' % domain
 
+@require_can_edit_commcare_users
```
Comment (dimagimon) [inline corehq/apps/users/views/mobile/users.py]: E303 too many blank lines (3)

File: corehq/apps/users/views/mobile/users.py [hunk 4]
```diff
@@ -1152,43 +1148,44 @@ def errors(self):
...
+def user_download_job_poll(request, domain, download_id, template="users/mobile/partials/user_download_status.html"):
```
Comment (dimagimon) [inline corehq/apps/users/views/mobile/users.py]: E501 line too long (117 > 115 characters)

**Pull request merged**